scripts/party.coffee

```diff
@@ -0,0 +1,55 @@
+# Description:
+#   Parties with two emoji
+#
+# Dependencies:
+#    none
+#
+# Configuration:
+#   None
+#
+# Commands:
+#    hubot party <emoji> <emoji> - ralph parties with your two emoji
+#    hubot party <emoji> <emoji> WxH - ralph parties with your two emoji in a W by H rectangle.
+#
+# Notes:
+#    None
+#
+# Author:
+#    @km42levelup
+
+DEFAULT_WIDTH = 12
+
+DEFAULT_HEIGHT = 6
+
+party = (emoji_one, emoji_two, width, height) ->
+    line_one = Array(Math.floor(width / 2) + 1).join(emoji_one + emoji_two)
+    if ((width % 2) == 1)
+        line_one += emoji_one
+
+    line_two = Array(Math.floor(width / 2) + 1).join(emoji_two + emoji_one)
+    if ((width % 2) == 1)
+        line_two += emoji_two
+
+    paragraph = ''
+    i = 0
+    while i < height
+        if ((i % 2) == 0)
+            paragraph += line_one + '\n'
+        else
+            paragraph += line_two + '\n'
+        i++
+    return paragraph
+
+
+module.exports = (robot) ->
+    robot.respond /party\s+(\S+)\s+(\S+)(\s+[\d]+x[\d]+$)?/i, (msg) ->
```

Comment:
We might want to validate that the first two matching groups are valid emoji:

```suggestion
    robot.respond /party\s+(:\S+:)\s+(:\S+:)(\s+[\d]+x[\d]+$)?/i, (msg) ->
```

Reply:
I'm not sure if this works, and I'm not sure how to check. In particular, if I send raw emoji through zulip, does it reach ralph as `:cat:` instead of `🐱`?

Reply:
on a similar note, i'd love to validate any `:custom_emoji:` is actually a real emoji in zulip but I highly suspect there is no easy way to check

Reply:
I'm going to guess that they're not translated to the unicode codepoints. But that does bring up a good point: if the `:...:` check is put in place, non-Zulip-supported unicode characters couldn't be used for this.

Reply:
Good points. This is pretty low-stakes stuff, so we might as well be lenient in what we accept!